File: day2/problem1.py
sample1 = open('./inputs/sample1.txt', 'r').read()
input1 = open('./inputs/input1.txt', 'r').read()


def solution(input_txt):
    points = {
        'AX': 4, 'AY': 8, 'AZ': 3,
        'BX': 1, 'BY': 5, 'BZ': 9,
        'CX': 7, 'CY': 2, 'CZ': 6,
    }

    lines = input_txt.split('\n')
    total_points = 0

    for line in lines:
        total_points += points[line[0] + line[2]]

    return total_points


print('sample:', solution(sample1))
print('solution:', solution(input1))

File: day2/problem1.js
const sample1 = require('fs').readFileSync(require('path').resolve(__dirname, './inputs/sample1.txt'), 'utf8');
const input1 = require('fs').readFileSync(require('path').resolve(__dirname, './inputs/input1.txt'), 'utf8');

const solution = (input) => {
  // A/X: Rock - B/Y: Paper - C/Z: Scissors
  const points = {
    'AX': 4, 'AY': 8, 'AZ': 3,
    'BX': 1, 'BY': 5, 'BZ': 9,
    'CX': 7, 'CY': 2, 'CZ': 6,
  };

  return input
    .split('\n')
    .reduce((acc, cur) => acc + points[cur[0] + cur[2]], 0);
};

console.log('sample:', solution(sample1));
console.log('solution:', solution(input1));
